src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/ParameterAssignmentCheck.java

```diff
@@ -70,6 +70,13 @@
  *     local -= 2;  // OK
  *     return local;
  *   }
+ *
+ *   IntPredicate obj = a -&gt; ++a == 12; // violation
+ *   IntBinaryOperator obj2 = (int a, int b) -&gt; {
+ *       a++;     // violation
+ *       b += 12; // violation
+ *       return a + b;
+ *   };
```

Comment:
please also add example for compliant code (create variable that refers to arg and modify it instead)

Reply:
Added such an example.